datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -774,6 +770,23 @@ protected void preConsumerPollHook() {
     }
   }
 
+  /**
+   * Pre commit hook for all operations that need to be performed before committing offsets.
+   */
+  protected void preCommitHook() { }
+
+  /**
+   * Post commit hook for all operations that need to be performed after committing offsets.
+   * @param success Indicates whether the commit attempt was successful or not.
+   */
+  protected void postCommitHook(boolean success) {
+    if (!success) {
```

Comment:
I think the `postCommitHook()` should not throw the exception because if the method is overridden the derived class and it fails to implement this logic then ab exception will never be thrown.
Instead the logic should remain the same as before and this operation can remain as a no-op.

Reply:
+1